src/main/java/uk/gov/hmcts/dts/fact/controllers/admin/AdminCourtPostcodeController.java

```diff
@@ -0,0 +1,107 @@
+import uk.gov.hmcts.dts.fact.config.security.Role;
+import uk.gov.hmcts.dts.fact.exception.InvalidPostcodeException;
+import uk.gov.hmcts.dts.fact.services.admin.AdminCourtPostcodeService;
+import uk.gov.hmcts.dts.fact.services.validation.ValidationService;
+
+import java.net.URI;
+import java.util.List;
+
+import static org.springframework.http.ResponseEntity.created;
+import static org.springframework.http.ResponseEntity.ok;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_SUPER_ADMIN;
+
+@RestController
+@RequestMapping(
+    path = "/admin/courts",
+    produces = {MediaType.APPLICATION_JSON_VALUE}
+)
+public class AdminCourtPostcodeController {
+    private final AdminCourtPostcodeService adminService;
+    private final ValidationService validationService;
+
+    @Autowired
+    public AdminCourtPostcodeController(AdminCourtPostcodeService adminService, ValidationService validationService) {
+        this.adminService = adminService;
+        this.validationService = validationService;
+    }
+
+    /**
+     * Retrieves postcodes handled by the court for civil service type
+     * @param slug Court slug
+     * @return A list of postcodes
+     */
+    @GetMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Find court postcodes by slug")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Successful", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role({FACT_ADMIN, FACT_SUPER_ADMIN})
+    public ResponseEntity<List<String>> getCourtPostcodes(@PathVariable String slug) {
+        return ok(adminService.getCourtPostcodesBySlug(slug));
+    }
+
+    /**
+     * Adds new postcodes to be handled by the court for civil service type (
+     * @param slug Court slug
+     * @param postcodes a list of postcodes to be added
+     * @return A list of postcodes created if successful, else return a list of invalid postcodes and a '400' response
+     */
+    @PostMapping(path = "/{slug}/postcodes")
+    @ApiOperation("Add new court postcodes")
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Postcodes created", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 400, message = "Invalid postcodes", response = String.class, responseContainer = "List"),
+        @ApiResponse(code = 401, message = "Unauthorized"),
+        @ApiResponse(code = 403, message="Forbidden"),
+        @ApiResponse(code = 404, message = "Not Found")
+    })
+    @Role(FACT_SUPER_ADMIN)
+    public ResponseEntity<List<String>> addCourtPostcodes(@PathVariable String slug, @RequestBody List<String> postcodes) {
+        final List<String> invalidPostcodes = validationService.validatePostcodes(postcodes);
```

Comment:
For my own interest here

How do we delineate between invalid postcodes and postcodes added on the return? Is it based on the 201/400 response? 

If we are checking for a 400 response to indicate invalid postcodes as a list, would there not also be another case where we would get a bad request separate to that? 

May need to consider returning a 200 for invalid, or a 201 for created. It's prob fine, but should we have a chat?

Reply:
For the 404 response, this could be related to how the court slug is not found (potentially) as well as for when mapit can't find a postcode...might be worth adding some context into the different bits in the description